packages/core/src/App/Containers/Layout/header/header.tsx

```diff
@@ -7,19 +7,26 @@ import DefaultHeader from './default-header.jsx';
 import DTraderHeader from './dtrader-header.jsx';
 import TradersHubHeader from './traders-hub-header';
 import DTraderHeaderWallets from './dtrader-header-wallets';
+import { useReadLocalStorage } from 'usehooks-ts';
 
 const Header = observer(() => {
     const { client } = useStore();
-    const { is_logged_in } = client;
+    const { is_logged_in, setAccounts, setLoginId } = client;
     const { pathname } = useLocation();
     const traders_hub_routes =
         [routes.traders_hub, routes.account, routes.cashier, routes.wallets, routes.compare_cfds].includes(pathname) ||
         pathname.startsWith(routes.compare_cfds);
 
+    const clientAccounts = useReadLocalStorage('client.accounts');
     const { is_next_wallet_enabled } = useFeatureFlags();
     const { has_wallet } = useStoreWalletAccountsList();
     const should_show_wallets = is_next_wallet_enabled && has_wallet;
 
+    React.useEffect(() => {
+        setAccounts(clientAccounts as Record<string, ReturnType<typeof useStore>['client']['accounts'][number]>);
+        setLoginId(String(localStorage.getItem('active_loginid')));
```

Comment:
We dont need to switchAccount ? 🤔

Reply:
@thisyahlen-deriv when the user creates new wallet we are switching to a new account, but the store thinks we didn't switch, and in this case I set active loginid in the store from the localStorage which we changed when switches to a new wallet

Reply:
So then this automaticallly switches the account ? 🤔

Reply:
@thisyahlen-deriv when I go to trader's hub from the wallets correct wallet is selected so suppose it's switched. But agree with you better to use switchAccount instead. I'll change it soon

Reply:
no i mean in the legacy code. switch account from clientstore

Reply:
but lets just use switchaccount to be safe